Baymax/Logger/LogsTableViewController.swift

```diff
@@ -0,0 +1,279 @@
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return section == 0 ? "Settings" : "Logs"
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return 1
+        case 1:
+            return logFiles.count
+        default:
+            return 0
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        return indexPath.section == 1
+    }
+    
+    override func tableView(_ tableView: UITableView, shouldBeginMultipleSelectionInteractionAt indexPath: IndexPath) -> Bool {
+        return indexPath.section == 1
+    }
+    
+    override func tableView(_ tableView: UITableView, didBeginMultipleSelectionInteractionAt indexPath: IndexPath) {
+        guard indexPath.section == 1 else { return }
+        if !isSharing {
+            toggleSharing()
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        
+        guard indexPath.section == 1, !isSharing else { return nil }
+        
+        return UISwipeActionsConfiguration(actions: [
+            
+            UIContextualAction(style: .destructive, title: "Delete", handler: { [weak self] (_, _, handler) in
+                
+                guard let self = self else { return }
+                
+                let alert = UIAlertController(title: "Delete File", message: "Are you sure you want to do this? You won't be able to retrieve the file at a later date.", preferredStyle: .alert)
+                alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { (_) in
+                    handler(false)
+                }))
+                alert.addAction(UIAlertAction(title: "Delete", style: .destructive, handler: { (_) in
+                    let fm = FileManager.default
+                    let file = self.logFiles[indexPath.row]
+                    do {
+                        try fm.removeItem(at: file.url)
+                        self.logFiles.remove(at: indexPath.row)
+                        self.tableView.reloadData()
+                        handler(true)
+                    } catch _ {
+                        handler(false)
+                    }
+                }))
+                self.present(alert, animated: true, completion: nil)
+            })
+        ])
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        switch indexPath.section {
+        case 0:
+            
+            let cell = tableView.dequeueReusableCell(withIdentifier: "switchRow", for: indexPath)
```

Comment:
Not required :)

Reply:
I'd rather not have force unwrapping, even though we both know this would never crash anyways!

Reply:
Ahh interesting, that's fine with me :D 
I actually think casting `as! MyTableViewCell` is one of the rare cases of a valid use of `!`.Because it should always be a specific type, or highlight the programmer's error.